soufi/tests/functional/test_functional.py

```diff
@@ -1,86 +1,266 @@
 # Copyright (c) 2021 Cisco Systems, Inc. and its affiliates
 # All rights reserved.
 
+import logging
+import tempfile
+from subprocess import run as run_cmd  # nosec
+
 import testtools
 
 from soufi import finder
 from soufi.finder import SourceType
 
+# Setup a module-level cache dict so that it can be warmed up and re-used
+# between tests
+FUNCTEST_CACHE = dict()
+
+# These tests will use the `memory_pickle` cache backend, which
+# serializes the cache payload before storing it; this should surface
+# any obvious problems with the other serializing cache backends
+# without forcing them as a dependency.
+
 
 class FunctionalFinderTests(testtools.TestCase):
     def setUp(self):
         self.skipTest("functional tests disabled for unit test runs")
```

Comment:
Maybe add a tox `[functional]` target to run these.

Reply:
Cool.  I was deferring any real action until I knew if you wanted these run with the main tests or not.  I figured you'd have opinions.  😆